### DISCUSSION TITLE
Parser for Trivy-Operator Prometheus metrics

### DISCUSSION THREAD

**Discussion by JuLiaN47V** [SecObserve/SecObserve#1505] [Ideas]
At this time, there is no way to parse data from the Prometheus-API.  

For example, i am running the Trivy-Operator on multiple Kubernetes-Clusters to generate Runtime-Vulnerability-Reports.  
These reports get fetched by prometheus as central datasource.  

The best way to get all reports across multiple clusters into SecObserve is to create a parser for Prometheus, containing the Trivy-Reports.  

I would contribute this parser to the project.  
What do you think?

**Comment by StefanFl** [comment]
Hi @JuLiaN47V, this sounds like a great idea. I would be happy to support you, if you can write the parser.

**Comment by JuLiaN47V** [comment]
Pull-Request: https://github.com/MaibornWolff/SecObserve/pull/1514  
I am not too familiar with TypeScript and React.  
At the API-Configuration-Add dialog, there are always all datafields displayed for every parser.  
Is it possible to only show the required datafields for each parser, based on the parser-selection?
